### PR TITLE
Typos: fix typos in comments

### PR DESCRIPTION
**Pull request by mykaul**
Fixes some typos as found by codespell run on the code. In this commit, I was hoping to fix only comments, not user-visible alerts, output, etc. Follow-up commits will take care of them.

Refs: https://github.com/scylladb/scylladb/issues/16255

### CODE REVIEW

File: cql3/expr/expression.cc [hunk 1]
```diff
@@ -1679,14 +1679,14 @@ cql3::raw_value do_evaluate(const binary_operator& binop, const evaluation_input
 }
 
 // Evaluate a conjunction of elements separated by AND.
-// NULL is treated as an "unkown value" - maybe true maybe false.
```
Comment (mykaul) [inline cql3/expr/expression.cc:1682]: That I'm not sure about - as there used to be a UKNOWN in C* once, but it was renamed at some point to UNKNOWN...

File: db/view/view.cc [hunk 2]
```diff
@@ -313,7 +313,7 @@ bool clustering_prefix_matches(data_dictionary::database db, const schema& base,
     auto selection = cql3::selection::selection::for_columns(base.shared_from_this(), ck_columns);
     uint64_t zero = 0;
     auto dummy_options = cql3::query_options({ });
-    // FIXME: pass nullptrs for some of theses dummies
+    // FIXME: pass nullptrs for some of  dummies
```
Comment (mykaul) [inline db/view/view.cc:316]:
```suggestion
    // FIXME: pass nullptrs for some of  these dummies
```

File: raft/tracker.hh [hunk 3]
```diff
@@ -38,7 +38,7 @@ public:
         PROBE,
         // In this state multiple append entries are sent optimistically
         PIPELINE,
-        // In this state snapshot is been transfered
+        // In this state snapshot is been transferred
```
Comment (mykaul) [inline raft/tracker.hh:41]:
```suggestion
        // In this state snapshot has been transferred
```

File: service/topology_state_machine.hh [hunk 4]
```diff
@@ -26,7 +26,7 @@
 namespace service {
 
 enum class node_state: uint16_t {
-    none,                // the new node joined group0 but did not bootstraped yet (has no tokens and data to serve)
+    none,                // the new node joined group0 but did not bootstrapped yet (has no tokens and data to serve)
```
Comment (mykaul) [inline service/topology_state_machine.hh:29]:
```suggestion
    none,                // the new node joined group0 but has not bootstrapped yet (has no tokens and data to serve)
```

**Review by nyh — APPROVED** [review]

**Comment by nyh** [comment]
Oh, I see now CI **did** pass, and then you only added a few small changes to comments and re-pushed. So I'll merge now.

File: raft/tracker.cc [hunk 5]
```diff
@@ -91,10 +91,10 @@ bool follower_progress::can_send_to() {
 }
 
 // If this is called when a tracker is just created, the current
-// progress is empty and we should simply crate an instance for
+// progress is empty and we should simply create an instance for
```
[the remark this answers is not in the document]
Comment (mykaul) [reply raft/tracker.cc:94]: Even more impressive as it is used regularly in Rust!

**Comment by mykaul** [comment]
@benipeled - any idea why it was aborted? I could not figure it out from the Jenkins logs.

**Comment by nyh** [comment]
Another CI failure that I don't understand (and it's not clear to me how to restart it).

**Comment by mykaul** [comment]
I've tried restart with 'rebuild' of the same job. Let's see what we get.

**Comment by fruch** [comment]
> fleet issues @fruch can you please take a look, something with the Jenkins<->AWS sync following today's upgrade

what exactly was upgraded ?

File: raft/tracker.cc (hunk 5 above)
[the remark this answers is not in the document]
Comment (tchaikov) [reply raft/tracker.cc:94]: > What spell-check did you use? Does it also do syntax analysis?

quote from the cover letter of this PR:

> Fixes some typos as found by codespell run on the code.

so, i think it is [codespell](https://github.com/codespell-project/codespell) .

**Comment by mykaul** [comment]
We can easily add a GH action for codespell. For example:
```
---
# codespell is a GitHub Actions that runs codespell tool to catch misspell
# Reference https://github.com/codespell-project/actions-codespell
name: Codespell
# yamllint disable-line rule:truthy
on:
  pull_request:
    branches:
      - '*'
permissions:
  contents: read

jobs:
  codespell:
    name: codespell
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: codespell
        run: make containerized-test TARGET=codespell
```

**Comment by tchaikov** [comment]
> We can easily add a GH action for codespell. For example:
> 
> ```
> ---
> # codespell is a GitHub Actions that runs codespell tool to catch misspell
> # Reference https://github.com/codespell-project/actions-codespell
> name: Codespell
> # yamllint disable-line rule:truthy
> on:
>   pull_request:
>     branches:
>       - '*'
> permissions:
>   contents: read
> 
> jobs:
>   codespell:
>     name: codespell
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v4
>       - name: codespell
>         run: make containerized-test TARGET=codespell
> ```

let's do it. see #16285

**Pull request merged**